tests/generic_views/test_base.py

```diff
@@ -286,6 +287,14 @@ def test_direct_instantiation(self):
         response = view.dispatch(self.rf.head('/'))
         self.assertEqual(response.status_code, 405)
 
+    def test_no_self_in_as_view_signature(self):
+        """
+        Test to check that as_view function's signature does not contain self.
+        """
```

Comment:
The method name is misleading as it is not the signature of `View.as_view()`, but that of the view function it generates. With a good method name we also don't need the docstring as it is clear enough:
```suggestion
    def test_self_not_in_signature_of_view_generated_by_as_view(self):
```

Quite long, but explicit. 😂

Reply:
Oh yes, it technically is the signature of the view function.

Haha yes!